src/Gom/main.js

```diff
@@ -0,0 +1,157 @@
+import GObject from "gi://GObject";
+import Gom from "gi://Gom";
+import Gtk from "gi://Gtk";
+import Adw from "gi://Adw";
+import GLib from "gi://GLib";
+import Gio from "gi://Gio";
+
+const INT32_MAX = 2147483647;
+
+const ItemClass = GObject.registerClass({
+  GTypeName: 'Item',
+  Properties: {
+    'id': GObject.ParamSpec.int(
+      'id', 'ID', 'An ID',
+      GObject.ParamFlags.READWRITE | GObject.ParamFlags.CONSTRUCT,
+      0, INT32_MAX, 0
+    ),
+    'url': GObject.ParamSpec.string(
+      'url', 'URL', 'A URL',
+      GObject.ParamFlags.READWRITE | GObject.ParamFlags.CONSTRUCT,
+      ''
+    ),
+  },
+}, class ItemClass extends Gom.Resource { });
+
+let adapter, repository;
+
+function initDatabase() {
+  adapter = new Gom.Adapter();
+  adapter.open_sync(':memory:',);
```

Comment:
```suggestion
  adapter.open_sync(':memory:', null);
```

Reply:
This method takes only a single parameter according to the console output 

`Gjs-Message: 18:11:55.995: JS WARNING: [file:///run/user/1000/re.sonny.Workbench.Devel/workbench-1723466515939 37]: Too many arguments to method Gom.Adapter.open_sync: expected 1, got 2
`

Reply:
Oh, my mistake. It's just one parameter, so you can remove the comma. I thought it took a cancellable, but the usual arg for that seems to be missing in most of the API.